Formula/php.rb

```diff
@@ -0,0 +1,258 @@
+      --with-pdo-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-png-dir=#{Formula["libpng"].opt_prefix}
+      --with-pspell=#{Formula["aspell"].opt_prefix}
+      --with-snmp
+      --with-tidy=shared,#{Formula["tidy-html5"].opt_prefix}
+      --with-unixODBC=#{Formula["unixodbc"].opt_prefix}
+      --with-webp-dir=#{Formula["webp"].opt_prefix}
+      --with-pic
+      --with-xmlrpc
+      --with-zlib=/usr
+      --with-libedit
+      --with-xsl=/usr
+    ]
+
+    if MacOS.version < :lion
+      args << "--with-curl=#{Formula["curl"].opt_prefix}"
+    else
+      args << "--with-curl"
+    end
+
+    args << "--enable-maintainer-zts" if build.with? "thread-safety"
+
+    system "./buildconf", "--force"
+    system "./configure", *args
+
+    inreplace "Makefile",
+      /^INSTALL_IT = \$\(mkinstalldirs\) '([^']+)' (.+) LIBEXECDIR=([^\s]+) (.+)$/,
+      "INSTALL_IT = $(mkinstalldirs) '#{libexec}/apache2' \\2 LIBEXECDIR='#{libexec}/apache2' \\4"
+
+    # https://github.com/phpbrew/phpbrew/commit/18ef766d0e013ee87ac7d86e338ebec89fbeb445
+    # Unsure if this is still needed
+    inreplace "Makefile" do |s|
+      s.change_make_var! "EXTRA_LIBS", "\\1 -lstdc++"
+    end
+
+    system "make"
+    ENV.deparallelize
+    system "make", "install"
+
+    bin.install_symlink "phar.phar" => "phar"
+
+    config_path.install "./php.ini-development" => "php.ini"
+
+    chmod 0755, "sapi/fpm/init.d.php-fpm"
+    sbin.install "sapi/fpm/init.d.php-fpm" => "php#{version.to_s[0..2].delete(".")}-fpm"
+
+    if !File.exist?(config_path/"php-fpm.d/www.conf") && File.exist?(config_path/"php-fpm.d/www.conf.default")
+      mv(config_path/"php-fpm.d/www.conf.default", config_path/"php-fpm.d/www.conf")
+    end
+
+    unless File.exist?(config_path/"php-fpm.conf")
+      config_path.install "sapi/fpm/php-fpm.conf"
+      inreplace config_path/"php-fpm.conf" do |s|
+        s.sub!(/^;?daemonize\s*=.+$/, "daemonize = no")
+      end
+    end
+  end
+
+  def caveats
+    s = []
+
+    s << <<-EOS.undent
+      To enable PHP in Apache add the following to httpd.conf and restart Apache:
+          LoadModule php7_module #{HOMEBREW_PREFIX}/opt/php#{version.to_s[0..2].delete(".")}/libexec/apache2/libphp7.so
+
+          <FilesMatch \.php$>
+              SetHandler application/x-httpd-php
+          </FilesMatch>
+
+      Finally, check DirectoryIndex includes index.php
+          DirectoryIndex index.php index.html
+    EOS
+
+    s << <<-EOS.undent
+      The php.ini file can be found in:
+          #{etc}/php/#{version.to_s.split(".")[0..1].join(".")}/php.ini
+    EOS
+
+    s.join "\n"
+  end
+
+  def post_install
+    (prefix/"var/log").mkpath
+    touch prefix/"var/log/php-fpm.log"
+
+    chmod 0755, lib/"php/.channels"
+    chmod 0755, lib/"php/.channels/.alias"
+    chmod 0644, (Dir.glob(lib/"php/.channels/**/*", File::FNM_DOTMATCH).reject { |a| a =~ %r{\/\.{1,2}$} || File.directory?(a) })
```

Comment:
The line is too long and I find it difficult to follow what's going on here.

Reply:
I need to create a list of files that includes files starting with a dot to change the permissions. glob includes . and .. when using the option FNM_DOTMATCH which then changes the permission on the directories which it shouldn't. The reject part filters out the . and .. as well as any directories in the tree. Any suggestions on what I can do instead ?